src/librustc_mir/borrow_check.rs

```diff
@@ -501,7 +513,9 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
                      lvalue_span: (&Lvalue<'tcx>, Span),
                      kind: ShallowOrDeep,
                      mode: MutateMode,
+                     is_local_mutation_allowed: LocalMutationIsAllowed,
```

Comment:
This function checks for local mutation anyway, so it should always pass `LocalMutationIsAllowed::Yes`